packages/abstract-signer/src.ts/index.ts

```diff
@@ -237,9 +235,10 @@ export abstract class Signer {
             // Prevent this error from causing an UnhandledPromiseException
             tx.to.catch((error) => {  });
         }
-
-
-        if (tx.gasLimit == null) {
+        // won't modify the present custom data
+        const customData = await tx.customData;
+        // FileCreate and FileAppend always carry a customData.fileChunk object
+        if (customData && !customData.fileChunk && tx.gasLimit == null) {
```

Comment:
I think that the check is wrong:
if we have `customData.fileChunk` we must ignore the `gasLimit`, BUT if we do not have `customData.fileChunk` we must check for `gasLimit`.

```TypeScript
if (!(customData && !customData.fileChunk) && tx.gasLimit == null) {
...
}
```
ps. double check my suggestion

Reply:
Final version: 
```
if (!(customData && customData.fileChunk) && tx.gasLimit == null) {
            // throw
        }
```